File: scripts/replace_domain.py
import os
import re

def replace_in_files(root_dir):
    """Recursively replace domain and email references in all text files"""
    updated_files = []
    
    # Define file extensions to process
    text_extensions = {'.tsx', '.ts', '.jsx', '.js', '.json', '.md', '.txt', '.xml', '.mjs', '.mts', '.html'}
    
    # Directories to skip
    skip_dirs = {'.next', '.git', 'node_modules', '.env', '.cache', 'dist', 'build'}
    
    for root, dirs, files in os.walk(root_dir):
        # Remove skip directories from dirs in-place to prevent traversal
        dirs[:] = [d for d in dirs if d not in skip_dirs]
        
        for file in files:
            # Check file extension
            _, ext = os.path.splitext(file)
            if ext not in text_extensions:
                continue
            
            file_path = os.path.join(root, file)
            
            try:
                with open(file_path, 'r', encoding='utf-8', errors='ignore') as f:
                    content = f.read()
                
                original_content = content
                
                # Replace domain references
                content = re.sub(r'luigioil\.com', 'www.luigiofficialbrand.com', content)
                
                # Replace email references
                content = re.sub(r'info@luigioil\.com', 'support@luigiofficialbrand.com', content)
                
                # If content changed, write it back
                if content != original_content:
                    with open(file_path, 'w', encoding='utf-8') as f:
                        f.write(content)
                    updated_files.append(file_path)
                    print(f"✓ Updated: {file_path}")
                    
            except Exception as e:
                print(f"✗ Error processing {file_path}: {e}")
    
    return updated_files

# Run the replacement
project_root = '/vercel/share/v0-project'

print(f"Starting domain and email replacement in {project_root}\n")
updated = replace_in_files(project_root)
print(f"\n✓ Successfully updated {len(updated)} files")

File: scripts/replace-domain.mjs
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const projectRoot = path.join(__dirname, '..');

function getAllFiles(dir, ext = null) {
  let results = [];
  const files = fs.readdirSync(dir);

  for (const file of files) {
    if (file.startsWith('.') || file === 'node_modules' || file === '.next' || file === '.git') continue;
    
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);

    if (stat.isDirectory()) {
      results = results.concat(getAllFiles(filePath, ext));
    } else if (!ext || filePath.endsWith(ext)) {
      results.push(filePath);
    }
  }
  return results;
}

function replaceInFile(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf-8');
    let modified = false;

    // Replace domain references
    if (content.includes('luigioil.com')) {
      content = content.replace(/luigioil\.com/g, 'www.luigiofficialbrand.com');
      modified = true;
    }

    // Replace email references
    if (content.includes('info@luigioil.com')) {
      content = content.replace(/info@luigioil\.com/g, 'support@luigiofficialbrand.com');
      modified = true;
    }

    if (modified) {
      fs.writeFileSync(filePath, content, 'utf-8');
      console.log(`✓ Updated: ${filePath}`);
      return true;
    }
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
  }
  return false;
}

// Get all files to process
const allFiles = getAllFiles(projectRoot);
const textFileExtensions = ['.tsx', '.ts', '.jsx', '.js', '.json', '.mjs', '.mts', '.md', '.txt', '.xml'];
const filesToProcess = allFiles.filter(file => 
  textFileExtensions.some(ext => file.endsWith(ext))
);

console.log(`Processing ${filesToProcess.length} files...\n`);

let updateCount = 0;
for (const file of filesToProcess) {
  if (replaceInFile(file)) {
    updateCount++;
  }
}

console.log(`\n✓ Successfully updated ${updateCount} files`);
